client/src/components/byline-autocomplete/index.jsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import Autocomplete from 'react-autocomplete';
+
+// Hooks.
+import useDebounce from '../../services/use-debounce';
+
+const BylineAutocomplete = ({
+  profiles,
+  onUpdate,
+  profilesApiUrl,
+  addAuthorPlaceholder,
+  addAuthorLabel,
+}) => {
+  const [search, setSearch] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+
+  // Debounce search string from input.
+  const debouncedSearchString = useDebounce(search, 750);
+
+  const doProfileSearch = (fragment) => {
+    fetch(
+      `${profilesApiUrl}?s=${fragment}`,
```

Comment:
Given that `@wordpress` scripts will be available, any reason to not use `@wordpress/url.addQueryArgs()` here so that `?` doesn't have to be manually concatenated and so that any existing query parameters in `profilesApiUrl` are accommodated?

Reply:
I guess I'm not sure if this is something we'd ever have to worry about. Since we're hitting these API URLs directly from the root. Though I get the idea. I might make a note of that in an issue for improvement for now since we're waiting on a few of these updates.

Reply:
https://github.com/alleyinteractive/byline-manager/issues/62